client/files.go

```diff
@@ -364,3 +368,135 @@ func (client *Client) RemovePath(opts *RemovePathOptions) error {
 
 	return nil
 }
+
+// PushOptions contains the options for a call to Push.
+type PushOptions struct {
+	// Source is the source of data to write (required).
+	Source io.Reader
+
+	// Path indicates the absolute path of the file in the destination
+	// machine (required).
+	Path string
+
+	// MakeDirs, if true, will create any non-existing directories in the path
+	// to the remote file. If false, the default, the call to Push will
+	// fail if any non-existing directory is found on the remote path.
+	MakeDirs bool
+
+	// Permissions indicates the mode of the file in the destination machine.
+	// Defaults to 0644. Note that, when used together with MakeDirs, the
+	// directories that might be created will not use this mode, but 0755.
+	Permissions string
```

Comment:
It seems to me this should be of type `os.FileMode`, like the similar `Permissions` field in `MakeDirOptions`. Can you remember why we made it a string here?

```suggestion
	// Permissions indicates the mode of the file on the destination machine.
	// If 0 or unset, defaults to 0644. Note that, when used together with MakeDirs, the
	// directories that might be created will not use this mode, but 0755.
	Permissions os.FileMode
```

Reply:
I wrote this long ago, so I don't really recall why I did this in the first place (: 
But that's a good catch! Thank you